--- origsrc/ruby-1.8.7-p71/ext/dl/handle.c	2007-02-12 17:01:19.000000000 -0600
+++ src/ruby-1.8.7-p71/ext/dl/handle.c	2008-08-15 03:08:38.950718800 -0500
@@ -10,9 +10,12 @@
 void
 dlhandle_free(struct dl_handle *dlhandle)
 {
+  if (!dlhandle)
+    return;
   if (dlhandle->ptr && dlhandle->open && dlhandle->enable_close) {
     dlclose(dlhandle->ptr);
   }
+  dlfree(dlhandle);
 }
 
 VALUE
--- origsrc/ruby-1.8.7-p71/ext/dl/ptr.c	2008-02-14 02:46:06.000000000 -0600
+++ src/ruby-1.8.7-p71/ext/dl/ptr.c	2008-08-15 03:08:38.935093900 -0500
@@ -45,6 +45,8 @@
 void
 dlptr_free(struct ptr_data *data)
 {
+  if (!data)
+    return;
   if (data->ptr) {
     DEBUG_CODE({
       printf("dlptr_free(): removing the pointer `0x%x' from the MemorySpace\n",
@@ -61,6 +63,7 @@
   if (data->stype) dlfree(data->stype);
   if (data->ssize) dlfree(data->ssize);
   if (data->ids) dlfree(data->ids);
+  dlfree(data);
 }
 
 void
--- origsrc/ruby-1.8.7-p71/ext/dl/sym.c	2007-07-22 00:33:47.000000000 -0500
+++ src/ruby-1.8.7-p71/ext/dl/sym.c	2008-08-15 03:08:38.950718800 -0500
@@ -57,6 +57,8 @@
 void
 dlsym_free(struct sym_data *data)
 {
+  if(!data)
+    return;
   if( data->name ){
     DEBUG_CODE({
       printf("dlsym_free(): free(data->name:%s)\n",data->name);
@@ -69,6 +71,7 @@
     });
     free(data->type);
   }
+  dlfree(data);
 }
 
 VALUE
