--- origsrc/ruby-1.9.3-p327/file.c	2012-11-09 10:50:39.000000000 -0600
+++ src/ruby-1.9.3-p327/file.c	2012-11-11 01:20:56.314518000 -0600
@@ -3190,7 +3190,7 @@ rb_file_expand_path_internal(VALUE fname
 	    len = lstrlenW(wfd.cFileName);
 #ifdef __CYGWIN__
 	    if (lnk_added && len > 4 &&
-		wcsicmp(wfd.cFileName + len - 4, L".lnk") == 0) {
+		wcscasecmp(wfd.cFileName + len - 4, L".lnk") == 0) {
 		wfd.cFileName[len -= 4] = L'\0';
 	    }
 #else
